### ISSUE TITLE
QC for BonoboGhost_4K19 (PanTro)

### ISSUE THREAD

**Issue by xin-huang** [popsim-consortium/stdpopsim#1227]
**PR for new model:** https://github.com/popsim-consortium/stdpopsim/pull/1215
**Original paper:** https://www.nature.com/articles/s41559-019-0881-7
**Parameter values:** Kuhlwilm et al. (2019) Supplementary Table S2.2 column I (Simplified model used in simulations) and Table S3 row 4 (Basic with 2% archaic flow into bonobo)
https://github.com/popsim-consortium/stdpopsim/blob/main/docs/parameter_tables/PanTro/BonoboGhost_4K19.csv
**Population indices:** 
0: Western Chimpanzees
1: Central Chimpanzees
2: Bonobos
3: An extinct ghost lineage

**Potential issues that might lead to differences between the model implementations:**
In Kuhlwilm et al. (2019) Supplementary Table S3, a population size change in bonobos after the split of bonobos and chimpanzees was modelled

<code>
ms 78 1250 -seeds [x x x] -t tbs -r tbs 250000.0 -I 4 22 36 20 0 -n 1 9.2 -n 2 65.9 -n 3 29.1 -n 4 10.0 -em 1.00075 1 2 60 -em 1.00075 2 1 20 -em 1.001 1 2 0 -em 1.001 2 1 0 -em 1.5505 3 2 4 -em 1.5505 2 3 5 -em 1.55075 3 2 0 -em 1.55075 2 3 0 -en 2.61 1 8 -en 3.08 3 3.7 -en 3.78 2 24.9 -em 5 3 4 80 -em 5.00025 3 4 0 -ej 7 1 2 -en 7 2 10.1 -em  12 3 2 0.0004 -em 12 2 3 0.0004 -em 15 2 3 0 -em 15 3 2 0 <b>-en 19.875 3 10.1</b> -ej 19.9 2 3 -en 19.9 3 11.6 -ej 35.0 3 4 -en 35.0 4 10.0
</code>

This should be population 2, the ancestors of chimpanzees, not of bonobos. And the population size should be 10.2 based on Supplementary Table S2.2 column I, assuming `N0=1000` in the above `ms` command.

**QC'er requests:**

**Comment by gtsambos** [comment]
Happy to give this a shot

**Comment by grahamgower** [comment]
Shameless plug: I see an ms command here, and [demes](https://github.com/popsim-consortium/demes-python/) has support for reading/converting ms commands. In conjunction with [demesdraw](https://github.com/grahamgower/demesdraw/) it's possible to do things like this:
```
$ demes ms <ms commands> | demesdraw tubes - <output-file.img>
```

E.g.
![BonoboGhost_4K19](https://user-images.githubusercontent.com/8665077/167794344-e916dd92-b74f-4eba-ae94-bb814a426139.svg)

**Comment by xin-huang** [comment]
Sorry, the link to the original paper was wrong. Updated now.

**Comment by xin-huang** [comment]
Hi @grahamgower 

how to specify sample size in the `ms -I` option when converting yaml files to `ms` command parameters?
I found a `samples` parameter in `demes.to_ms(graph, *, N0, samples=None)`  but it is unclear that how to use this parameter

**Comment by xin-huang** [comment]
and looks like the time unit in the figure above is years, but the y-label still used generations @grahamgower

**Comment by grahamgower** [comment]
> Hi @grahamgower
> 
> how to specify sample size in the `ms -I` option when converting yaml files to `ms` command parameters? I found a `samples` parameter in `demes.to_ms(graph, *, N0, samples=None)` but it is unclear that how to use this parameter

When converting a yaml file using the demes command line interface, there is no way to specify the sampling configuration. You'll have to manually edit the `-I` arguments to specify the samples you want when using the CLI. Using the Python API and the `demes.to_ms()` function, the `samples` argument is a list of integer samples. Please open an issue at https://github.com/popsim-consortium/demes-python/issues if you think this could be improved.

> and looks like the time unit in the figure above is years, but the y-label still used generations @grahamgower

I probably used the wrong `N0` value to convert from coalescent units. The yaml output from parsing an ms command will always be generations, but the actual values depend on having a sane `N0` (which I see you wrote in your original post, but I think I missed that before).

**Comment by xin-huang** [comment]
Hi @petrelharp 

Can I suggest @kuhlwilm to QC this model, because he is the author for this model?

**Comment by petrelharp** [comment]
> Can I suggest @kuhlwilm to QC this model, because he is the author for this model?

Yes, although they should make sure to QC based on what's written down in the paper, rather than private knowledge.

**Comment by petrelharp** [comment]
Note: I hear that @kuhlwllm has agreed to review this, but I can't "assign" them for some reason.

**Comment by petrelharp** [comment]
Ping @kuhlwllm?

**Comment by xin-huang** [comment]
> Ping @kuhlwllm?

He is on vacation now and already implemented https://github.com/kuhlwilm/stdpopsim/blob/main/stdpopsim/qc/PanTro.py
But he also suggested we should use the model based on ABC in the same paper instead of SFS (this model I implemented), because the model based on ABC fit data better. Should he also change the parameters in catalog (https://github.com/kuhlwilm/stdpopsim/blob/main/stdpopsim/catalog/PanTro/demographic_models.py)?

**Comment by petrelharp** [comment]
Ah, if it's already QC'ed, you could put his code in a PR? But, if you're going to switch over to the ABC model, I think we need two different people to switch over the main model and the QC model - up to you.

**Comment by xin-huang** [comment]
> Ah, if it's already QC'ed, you could put his code in a PR? But, if you're going to switch over to the ABC model, I think we need two different people to switch over the main model and the QC model - up to you.

So I should open a pr and change the parameters based on the results from ABC, then ask @kuhlwilm to open the pr for the QC because his QC is based on the ABC results?

**Comment by petrelharp** [comment]
Sounds good!

**Comment by xin-huang** [comment]
Hi @petrelharp 
I updated parameters in https://github.com/popsim-consortium/stdpopsim/pull/1350

**Comment by petrelharp** [comment]
OK, @kuhlwilm - your turn, for the QC, I believe.

**Comment by nspope** [comment]
Closed by #1370.
